pandas/tests/io/json/test_pandas.py

```diff
@@ -82,489 +80,379 @@ def setup(self, datapath):
+            [[1.5, 2.5], [3.5, 4.5]],
+            [[1, 2.5], [3, 4.5]],
+            [[Timestamp("20130101"), 3.5], [Timestamp("20130102"), 4.5]],
+        ],
+    )
+    def test_frame_non_unique_columns(self, orient, data):
+        df = DataFrame(data, index=[1, 2], columns=["x", "x"])
+
+        result = read_json(
+            df.to_json(orient=orient), orient=orient, convert_dates=["x"]
+        )
+        if orient == "values":
+            expected = pd.DataFrame(data)
+            if expected.iloc[:, 0].dtype == "datetime64[ns]":
+                expected.iloc[:, 0] = expected.iloc[:, 0].astype(np.int64) // 1000000
+        elif orient == "split":
+            expected = df
+
+        assert_frame_equal(result, expected)
 
-    def test_frame_non_unique_columns(self):
+    @pytest.mark.parametrize("orient", ["index", "columns", "records"])
+    def test_frame_non_unique_columns_raises(self, orient):
         df = DataFrame([["a", "b"], ["c", "d"]], index=[1, 2], columns=["x", "x"])
 
-        msg = "DataFrame columns must be unique for orient='index'"
-        with pytest.raises(ValueError, match=msg):
-            df.to_json(orient="index")
-        msg = "DataFrame columns must be unique for orient='columns'"
-        with pytest.raises(ValueError, match=msg):
-            df.to_json(orient="columns")
-        msg = "DataFrame columns must be unique for orient='records'"
+        msg = "DataFrame columns must be unique for orient='{}'".format(orient)
         with pytest.raises(ValueError, match=msg):
-            df.to_json(orient="records")
+            df.to_json(orient=orient)
 
-        assert_frame_equal(
-            df, read_json(df.to_json(orient="split"), orient="split", dtype=False)
+    def test_frame_default_orient(self):
+        assert self.frame.to_json() == self.frame.to_json(orient="columns")
+
+    @pytest.mark.parametrize("dtype", [False, float])
+    @pytest.mark.parametrize("convert_axes", [True, False])
+    @pytest.mark.parametrize("numpy", [True, False])
+    def test_roundtrip_simple(self, orient, convert_axes, numpy, dtype):
+        data = self.frame.to_json(orient=orient)
+        result = pd.read_json(
+            data, orient=orient, convert_axes=convert_axes, numpy=numpy, dtype=dtype
         )
-        unser = read_json(df.to_json(orient="values"), orient="values")
-        tm.assert_numpy_array_equal(df.values, unser.values)
 
-        # GH4377; duplicate columns not processing correctly
-        df = DataFrame([["a", "b"], ["c", "d"]], index=[1, 2], columns=["x", "y"])
-        result = read_json(df.to_json(orient="split"), orient="split")
-        assert_frame_equal(result, df)
+        expected = self.frame.copy()
 
-        def _check(df):
-            result = read_json(
-                df.to_json(orient="split"), orient="split", convert_dates=["x"]
-            )
-            assert_frame_equal(result, df)
+        if not numpy and (orient == "index" or (PY35 and orient == "columns")):
```

Comment:
not necesarily for this PR since its the same in master, but id prefer if this was e.g. "use_numpy" instead of "numpy", harder to confuse with the "someone didn't alias to `np`"

Reply:
Yea makes sense. I actually have to dive a little further on this keyword anyway as it's not immediately apparent to me what the use cases are, nor are any really documented AFAICT